#6969 - MODS export

mcantelon opened the pull request:
Addition of MODS bulk export, addition of new MODS export fields, and addition of a site base URL DB setting.

mcantelon committed:
Added ability to bulk export in MODS format.

Added the ability to select export format with the bulk export tool
(either "ead" or "mods").

mcantelon committed:
Fixed minor issues in MODS XML export template.

mcantelon committed:
Added parent relatedItem to MODS template.

mcantelon committed:
Added CLI arg for site URL to EAD bulk export.

CLI tasks have no knowledge of hostname, etc., so can't automatically
generate a full base URL for exported XML. Added a CLI arg to the
EAD bulk export tool to specify a URL. Integrated this into the existing
EAD template.

mcantelon committed:
Added/revise fields in the MODS XML template.

mcantelon committed:
Added access rights to MODS template.

Added support for indicating if access rights exist, fixed a bug in the
MODS template, and added rights-related helpered to MODS export class.

mcantelon committed:
Added replicate rights to MODS template.

Added support for indicating if replication restrictions exist in
MODS export.

mcantelon committed:
Added alphanumeric notes to MODS export.

Refactored and add the RAD alphanumeric designations note to MODS
export.

Commit:
Base URL now optional, change EAD identifier
The EAD identifier attribute in <eadid> now uses the top
level information object's slug instead of url_to().

mcantelon committed:
Added site base URL configuration parameter.

Added a site base URL configuration parameter (to both the
settings page and the purge task). Cleaned up the purge
task code.`

Added URI identifier to MODS export.

mcantelon committed:
Changed order of MODS export repository fields.

mcantelon committed:
Changed MODS bulk export so children export.

Changed the query for information object so if an export's format is
MODS, and the current level only option isn't flagged, child
descriptions are exported in addition to top-level descriptions.

Removed typo in MOD XML template and changed template so publication status
doesn't get shown in title of parents/children.

Updated relatedItem XML to conform to MODS standard.

mcantelon committed:
Added XML tidying to bulk export.

mcantelon committed:
Added display of total descriptions exported.

mcantelon committed:
Fixed issues with MODS export of digital objects.

Fixed issues with digital object paths in MODS export.

mcantelon committed:
Fixed issue with CLI bulk export breaking perms.

Fixed an issue with the CLI bulk export that caused permissions
not to work.

mcantelon committed:
Fixed issue with EAD export digital object URLs.

mcantelon committed:
Fix for repository name in MODS export.

Fix provided by Tim Hutchinson.

mcantelon committed:
Support non-clean URLs in MODS/EAD export.

Made the URLs exported by the MODS and EAD export work if clean URLs
aren't enabled.

sevein commented:
@mcantelon did you see #52? In particular, 11938bdd6af932ae1942a15a9e77cce8dfa93833.

mcantelon commented:
Hi Jesus,

Yeah, I'm pretty sure I made that fix manually, using his PR for
reference... sorry if I forgot to close the PR.

Cheers,
Mike

On Tue, Sep 30, 2014 at 4:37 PM, Jesús García Crespo <
notifications@github.com> wrote:

> @mcantelon https://github.com/mcantelon did you see #52
> https://github.com/artefactual/atom/pull/52?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/artefactual/atom/pull/55#issuecomment-57399621.

## 

Mike Cantelon ::  Open Source Development > LAMP / node.js / HTML5
http://mikecantelon.com

mcantelon committed:
Minor cleanup of MODS XML template.

Did some minor cleanup of the MODS XML export template and added padding
to the ID part of the exported file's name so they can be sorted by
creation (to make imports, etc.) easier.

mcantelon committed:
Fixed issues with MODS export template.

Fixed a number of issues with the MODS export that prevented exports
from validating.

MikeFE commented on lib/task/exportBulkTask.class.php:
```diff
@@ -139,15 +144,32 @@ protected function checkForValidFolder($folder)
+  {
+    // EAD data nests children, so we only have to get top-level items
+    $whereClause = ($options['format'] == 'ead' || $options['current-level-only'])
+      ? "parent_id=1"
```
Can we use QubitInformationObject::ROOT_ID instead of the magic number 1 here?

MikeFE commented on lib/task/exportBulkTask.class.php:
```diff
@@ -87,15 +92,31 @@ public function execute($arguments = array(), $options = array())
+      if ($options['format'] == 'ead')
+      {
+        $ead = new sfEadPlugin($resource);
+      } else {
```
I actually like this way of doing else statements better than what we usually use...

if (...)
{
}
else
{
}

But we should be consistent across the code base I think. I guess it's sort of moot at this point since I've seen a few different if/else styles :(

mcantelon committed:
Minor code style tweaks.

MikeFE commented:
I reviewed the code and Mike C has merged it (https://github.com/artefactual/atom/commit/21c9e82a96480dd736b9147c7488d53b16adfb74)
